src/data/types.d.ts

```diff
@@ -8,65 +11,318 @@ export type ProductSlug =
+
+  /**
+   * An array of items describing the primary top-level navigation for this
+   * product in Dev Portal.
+   * @see {@link NavigationHeaderItem}
+   */
+  navigationHeaderItems: NavigationHeaderItem[]
+
+  /**
+   * The current release version for this product.
+   */
   version?: string
+
+  /**
+   * The current release version for the Vagrant Vmware utility.
+   * (Vagrant Specific)
+   */
   vmwareUtilityVersion?: string
+
+  /**
+   * The URL to a product's changelog. Used on the /downloads page
+   */
   changelogUrl?: string
+
+  /**
+   * The configuration for the packageManager prop for a product's /download
+   * page. No longer used.
+   * @deprecated
+   * @see {@link PackageManager}
+   */
   packageManagers?: PackageManager[]
+
+  /**
+   * A read-only Dato token used to query for product-specific data.
+   */
   datoToken?: string
 }
 
 export interface AlertBanner {
+  /**
+   * The text contained inside the tag to the left
+   */
   tag: string
+
+  /**
+   * where to link to when clicked
+   */
   url: string
+
+  /**
+   * The text in the main area
+   */
   text: string
+
+  /**
+   * Secondary text styled as a link
+   */
   linkText: string
+
+  /**
+   * A datetime string that, when set, controls if the alert banner should appear
+   */
   expirationDate: Date
 }
 
 export interface AlgoliaConfig {
+  /**
+   * The name of the Algolia index to query for this product, usually in the
+   * form of product_PRODUCTSLUG.
+   */
   indexName: string
+
+  /**
+   * A search-only Algolia API key
+   */
   searchOnlyApiKey: string
 }
 
 export interface AnalyticsConfig {
+  /**
+   * A list of space-separated fully-qualified domain names for which to gather
+   * analytics.
+   */
   includedDomains: string
+
+  /**
+   * A Segment API key with write permissions.
+   */
   segmentWriteKey: string
 }
 
 export interface Metadata {
+  /**
+   * The text to use as the defaut content for the <title> element for pages
+   * using this product configuration.
+   */
   title: string
+
+  /**
+   * The text to use as the defaut content for the description element for
+   * pages using this product configuration.
+   */
   description: string
+
+  /**
+   * The URL to an image to use in OpenGraph and Twitter meta tags.
+   */
   image: string
+
+  /**
+   * A list of Icons to include in the <head> element.
+   * @see {@link Icon}
+   */
   icon: Icon[]
 }
 
 export interface Icon {
+  /**
+   * The URL to an image to use as a favicon
+   */
   href: string
+
+  /**
+   * The mimetype for the image located at href.
+   */
+  type?: string
+
+  /**
+   * The dimensions of the image located at href in WxH form.
+   */
+  sizes?: string
+}
+
+export interface NavigationHeaderItem {
+  /**
+   * The text used for this navigation item.
+   */
+  label: string
+
+  /**
+   * The relative path for this navigation item.
+   */
+  path: string
+
+  /**
+   * A unique identifier used as a key for React.
+   */
+  id: string
```

Comment:
Same as previous comment, this is already defined elsewhere and I'll handle documenting it in my other task. 😊